hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -221,7 +233,12 @@ private ResponseCodeEnum validate(TransactionBody cryptoUpdateTxn) {
 		if (op.hasAutoRenewPeriod() && !validator.isValidAutoRenewPeriod(op.getAutoRenewPeriod())) {
 			return AUTORENEW_DURATION_NOT_IN_RANGE;
 		}
-
+		if (op.hasProxyAccountID() && !op.getProxyAccountID().equals(AccountID.getDefaultInstance())) {
+			return PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED;
+		}
+		if (!validator.isValidStakedIdIfPresent(op.getStakedAccountId(), op.getStakedNodeId(), accounts.get())) {
```

Comment:
Imo this should also pass the `NodeInfo` singleton so we can validate the `nodeId`, if that option is chosen.

Reply:
Passed NodeInfo to the 4 transition logics and validated using `nodeInfo.accountOf(nodeId)`